environments/docker/docker.go

```diff
@@ -84,6 +84,8 @@ func (d *docker) dockerExecuteAsync(steps pufferpanel.ExecutionData) error {
 		return errors.New("docker container already exists")
 	}
 
+	d.ImageName = pufferpanel.ReplaceTokens(d.ImageName, steps.Data)
```

Comment:
I still don't like the idea of overriding the value, because this would persist.